File: gto-support-api-retrofit2/src/main/java/org/ccci/gto/android/common/api/retrofit2/converter/LocaleConverterFactory.java
package org.ccci.gto.android.common.api.retrofit2.converter;

import static org.ccci.gto.android.common.retrofit2.converter.LocaleConverterFactory.INSTANCE;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

import retrofit2.Converter;
import retrofit2.Retrofit;

/**
 * @deprecated Since v3.10.0, use {@link org.ccci.gto.android.common.retrofit2.converter.LocaleConverterFactory} instead
 */
@Deprecated
public class LocaleConverterFactory extends Converter.Factory {
    @Override
    public Converter<?, String> stringConverter(Type type, Annotation[] annotations, Retrofit retrofit) {
        return INSTANCE.stringConverter(type, annotations, retrofit);
    }
}

File: gto-support-retrofit2/src/main/kotlin/org/ccci/gto/android/common/retrofit2/converter/LocaleConverterFactory.kt
package org.ccci.gto.android.common.retrofit2.converter

import java.lang.reflect.Type
import java.util.Locale
import retrofit2.Converter
import retrofit2.Retrofit

object LocaleConverterFactory : Converter.Factory() {
    override fun stringConverter(
        type: Type,
        annotations: Array<Annotation>,
        retrofit: Retrofit
    ): Converter<*, String>? = when (type) {
        Locale::class.java -> LocaleStringConverter
        else -> null
    }

    private object LocaleStringConverter : Converter<Locale, String> {
        override fun convert(locale: Locale): String? = locale.toLanguageTag()
    }
}
